api/all/src/main/java/io/opentelemetry/api/internal/ValidationUtil.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.api.internal;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.Immutable;
+
+/** General internal validation utility methods. */
+@Immutable
+public final class ValidationUtil {
+
+  private ValidationUtil() {}
+
+  private static final Logger API_USAGE_LOGGER =
+      Logger.getLogger("io.opentelemetry.ApiUsageLogging");
+
+  public static void log(String msg) {
+    API_USAGE_LOGGER.log(Level.FINEST, msg, new NullPointerException());
```

Comment:
```suggestion
    API_USAGE_LOGGER.log(Level.FINEST, msg, new AssertionError());
```

Since we use in some cases for non-null condition, let's use `AssertionError` instead